docs/src/Typeahead.doc.js

```diff
@@ -0,0 +1,196 @@
+import PropTable from './components/PropTable.js';
+import PageHeader from './components/PageHeader.js';
+
+const cards = [];
+const card = c => cards.push(c);
+
+card(
+  <PageHeader
+    name="Typeahead"
+    description={`Use a \`Typeahead\` when you want to let the user filter a list when selecting.`}
+  />
+);
+
+card(
+  <PropTable
+    props={[
+      {
+        name: 'data',
+        type: 'Array<{ label: string, value: string }>',
+        description:
+          'The data much be an array with objects containing only label and value properties',
+        required: true,
+        href: 'basicExample',
+      },
+      {
+        name: 'defaultItem',
+        type: '{ label: string, value: string }',
+        description: 'The default item set in the Typeahead',
+        required: false,
+        href: 'defaultItemExample',
+      },
+      {
+        name: 'id',
+        type: 'string',
+        required: true,
+        href: 'basicExample',
+      },
+      {
+        name: 'label',
+        type: 'string',
+      },
+      {
+        name: 'noResultText',
+        type: 'string',
+        href: 'basicExample',
+        required: true,
+        description: 'The text shown when the input value returns no matches',
+      },
+      {
+        name: 'onBlur',
+        type:
+          '({ event: SyntheticFocusEvent<HTMLInputElement> | SyntheticEvent<HTMLInputElement> , value: string }) => void',
+        required: false,
+        description: 'Callback when you focus outside the component ',
+        href: 'basicExample',
+      },
+
+      {
+        name: 'onChange',
+        type: '({ event: SyntheticInputEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when user types into the control input field',
+        href: 'basicExample',
+      },
+
+      {
+        name: 'onFocus',
+        type: '({ event: SyntheticFocusEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when you focus on the component',
+        href: 'basicExample',
+      },
+      {
+        name: 'onSelect',
+        type: '({ event: SyntheticInputEvent<>, item: Object }) => void',
+        required: false,
+        description: 'Callback when you select an item ',
+        href: 'basicExample',
+      },
+      {
+        name: 'placeholder',
+        type: 'string',
+        href: 'basicExample',
+      },
+      {
+        name: 'searchField',
+        type: 'string',
+        description:
+          'The property name on which you want to be able to search on',
+        href: 'basicExample',
+        defaultValue: 'label',
+      },
+
+      {
+        name: 'size',
+        type: '"md" | "lg"',
+        required: false,
+        description: 'md: 40px, lg: 48px',
+        defaultValue: 'md',
+      },
+    ]}
+  />
+);
+
+card(
+  <Example
+    id="basicExample"
+    name="Example"
+    defaultCode={`
+    function Example(props) {
```

Comment:
@simkessy could we fix the indentation  of the examples?

`function Example(props) {` instead of `    function Example(props) {`.